qemu/target-arm/unicorn_arm.c

```diff
@@ -9,8 +9,9 @@
 #include "unicorn_common.h"
 #include "uc_priv.h"
 
-
+#ifndef TARGET_WORDS_BIGENDIAN
```

Comment:
I did it, since target-mips/unicorn.c had do a similar one.
The root cause is:  both little-endian and big-endian are using the same global definition. Then, at the link stage, compiler would report error as redefinition.

Maybe, we can define different REGS_STORAGE_SIZE for different endian, it may looks more beauty.